### ISSUE TITLE
Local Cache Whitelisting and Filtering

### ISSUE THREAD

**Issue by BlueStalker** [Alluxio/alluxio#13818]
**Is your feature request related to a problem? Please describe.**
Given a limited total Cap of Alluxio local cache per node, there is a need for only caching a specific set of tables with a specific set of partitions, based on the actual client requirement/config.

**Describe the solution you'd like**
The proposal would be adding another interface to determine whether needs to cache the file/split.
Additional config required to init the local Cache File System object (I suppose to) in order to make such information queryable. Metrics are also needed in terms of how many are going to be cache or not.

**Describe alternatives you've considered**
There are multiple alternative solutions.
1. Cache Quota support, while in a large production environment there could be over 10k or 20k tables, applying NO_QUOTA to filter is quite complicated and also it can not handle the case for the new dataset.
2. Potentially auto-tune support, which means there could be a threshold or some other mechanism in the cached inventory to ONLY cache dataset/files/partitions if the hit ratio is a certain point (X/hour, X/day, etc.) This is a complicated model which also has pros/cons, for example. How to deal with some spiky traffic and so, which leads to more concrete design.

In short term, I would still prefer to apply a filter.

**Urgency**
It is critical in our presto environment to use the local cache solution because we want to use this feature in order to efficiently use the limited caching quota.

**Additional context**
N/A
